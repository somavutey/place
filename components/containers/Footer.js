import { Grid, Typography } from "@mui/material";
import { makeStyles } from "@mui/styles";
import { activities, region, kindofplace } from "../../web-admin/_mock_/footer";
import Link from "next/link";
import FooterListComponent from "../presentations/FooterComponent";
import React from "react";
const useStyles = makeStyles({
  footer: {
    background: "#15A2B8",
    marginTop: 50,
    width: "100%",
    textAlign: "center",
    color: "white",
    fontSize: 18,
  },
  footerTitle: {
    fontSize: 18,
    marginTop: "20px",
    fontWeight: "bold",
  },
  footerSub: {
    display: "block",
    fontSize: "15px",
    lineHeight: "2em",
    "&:hover": {
      textDecoration: "underline",
    },
  },
  contact: {
    color: "#000",
    fontSize: 15,
  },
});

const Footer = () => {
  const classes = useStyles();
  return (
    <div>
      <div className={classes.footer}>
        <Grid container spacing={2}>
          {/* - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
                                                         List all Acitivities 
                                                      
              - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -*/}
          <Grid item sm={4} xs={12}>
            <Typography variant="h3" className={classes.footerTitle}>
              More activities
            </Typography>

            {activities.map((items, index) => {
              return (
                <FooterListComponent key={index}>
                  {items.item}
                </FooterListComponent>
              );
            })}
          </Grid>
          <Grid item sm={4} xs={12}>
            <Typography variant="h3" className={classes.footerTitle}>
              More activities
            </Typography>
            {kindofplace.map((items, index) => {
              return (
                <FooterListComponent key={index}>
                  {items.item}
                </FooterListComponent>
              );
            })}
          </Grid>
          <Grid item sm={4} xs={12}>
            <Typography variant="h3" className={classes.footerTitle}>
              More activities
            </Typography>
            {region.map((items, index) => {
              return (
                <FooterListComponent key={index}>
                  {items.item}
                </FooterListComponent>
              );
            })}
          </Grid>
          {/* - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
                                                         Draw Horizontal Line                    
              - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -*/}
          <Grid item lg={12} sm={12} xs={12}>
            <hr
              style={{ border: "none", height: 3, backgroundColor: "black" }}
            />
          </Grid>
          {/* - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
                                                         Contact Us and About Us                    
              - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -*/}
          <Grid item textAlign="center" lg={6} sm={6} xs={6}>
            <Typography
              variant="h5"
              color="#000"
              style={{ fontWeight: "bold", fontSize: "clamp(1rem,1.5vw,2rem)" }}
            >
              CONTACT US!
            </Typography>
          </Grid>
          <Grid item textAlign="center" lg={6} sm={6} xs={6}>
            <Link href="#">
              <a
                style={{
                  display: "block",
                  color: "#000",
                  fontSize: "clamp(1rem,1.5vw,2rem)",
                  fontWeight: "bold",
                  textDecoration: "underline",
                  lineHeight: 1.235,
                }}
              >
                ABOUT US!
              </a>
            </Link>
          </Grid>

          <Grid container style={{ margin: "50px 100px" }}>
            <Grid item md={4} sm={4} xs={12}>
              <Typography className={classes.contact}>
                2007st Kakab, Khan Puthisen Chey, Phnom Penh
              </Typography>
            </Grid>
            <Grid item md={2} sm={4} xs={12}>
              <Typography className={classes.contact}>
                +855 10 46 61 47
              </Typography>
            </Grid>
            <Grid item md={2} sm={4} xs={12}>
              <Typography className={classes.contact}>@Place</Typography>
            </Grid>
          </Grid>
        </Grid>
      </div>
    </div>
  );
};

export default Footer;
