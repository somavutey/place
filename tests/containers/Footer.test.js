import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Footer from "./../../components/containers/Footer";

it("Footer", () => {
  const wrapper = shallow(<Footer />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
