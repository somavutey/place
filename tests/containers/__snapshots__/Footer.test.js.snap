// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer 1`] = `
<div>
  <div
    className="makeStyles-footer-1"
  >
    <ForwardRef(Grid)
      container={true}
      spacing={2}
    >
      <ForwardRef(Grid)
        item={true}
        sm={4}
        xs={12}
      >
        <ForwardRef(Typography)
          className="makeStyles-footerTitle-2"
          variant="h3"
        >
          More activities
        </ForwardRef(Typography)>
        <FooterListComponent
          key="0"
        >
          Shopping
        </FooterListComponent>
        <FooterListComponent
          key="1"
        >
          Sightseeing
        </FooterListComponent>
        <FooterListComponent
          key="2"
        >
          Eating & Drinkning
        </FooterListComponent>
        <FooterListComponent
          key="3"
        >
          Learning 
        </FooterListComponent>
        <FooterListComponent
          key="4"
        >
          Entertaining
        </FooterListComponent>
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
        sm={4}
        xs={12}
      >
        <ForwardRef(Typography)
          className="makeStyles-footerTitle-2"
          variant="h3"
        >
          More activities
        </ForwardRef(Typography)>
        <FooterListComponent
          key="0"
        >
          Modern
        </FooterListComponent>
        <FooterListComponent
          key="1"
        >
          History
        </FooterListComponent>
        <FooterListComponent
          key="2"
        >
          Outdoor
        </FooterListComponent>
        <FooterListComponent
          key="3"
        >
          Indoor
        </FooterListComponent>
        <FooterListComponent
          key="4"
        >
          Nature
        </FooterListComponent>
        <FooterListComponent
          key="5"
        >
          Event
        </FooterListComponent>
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
        sm={4}
        xs={12}
      >
        <ForwardRef(Typography)
          className="makeStyles-footerTitle-2"
          variant="h3"
        >
          More activities
        </ForwardRef(Typography)>
        <FooterListComponent
          key="0"
        >
          Kampot
        </FooterListComponent>
        <FooterListComponent
          key="1"
        >
          Sihanouk Ville
        </FooterListComponent>
        <FooterListComponent
          key="2"
        >
          Koh Kong
        </FooterListComponent>
        <FooterListComponent
          key="3"
        >
          Kep
        </FooterListComponent>
        <FooterListComponent
          key="4"
        >
          Kratie
        </FooterListComponent>
        <FooterListComponent
          key="5"
        >
          Kampong Cham
        </FooterListComponent>
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
        lg={12}
        sm={12}
        xs={12}
      >
        <hr
          style={
            Object {
              "backgroundColor": "black",
              "border": "none",
              "height": 3,
            }
          }
        />
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
        lg={6}
        sm={6}
        textAlign="center"
        xs={6}
      >
        <ForwardRef(Typography)
          color="#000"
          style={
            Object {
              "fontSize": "clamp(1rem,1.5vw,2rem)",
              "fontWeight": "bold",
            }
          }
          variant="h5"
        >
          CONTACT US!
        </ForwardRef(Typography)>
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
        lg={6}
        sm={6}
        textAlign="center"
        xs={6}
      >
        <Link
          href="#"
        >
          <a
            style={
              Object {
                "color": "#000",
                "display": "block",
                "fontSize": "clamp(1rem,1.5vw,2rem)",
                "fontWeight": "bold",
                "lineHeight": 1.235,
                "textDecoration": "underline",
              }
            }
          >
            ABOUT US!
          </a>
        </Link>
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        container={true}
        style={
          Object {
            "margin": "50px 100px",
          }
        }
      >
        <ForwardRef(Grid)
          item={true}
          md={4}
          sm={4}
          xs={12}
        >
          <ForwardRef(Typography)
            className="makeStyles-contact-4"
          >
            2007st Kakab, Khan Puthisen Chey, Phnom Penh
          </ForwardRef(Typography)>
        </ForwardRef(Grid)>
        <ForwardRef(Grid)
          item={true}
          md={2}
          sm={4}
          xs={12}
        >
          <ForwardRef(Typography)
            className="makeStyles-contact-4"
          >
            +855 10 46 61 47
          </ForwardRef(Typography)>
        </ForwardRef(Grid)>
        <ForwardRef(Grid)
          item={true}
          md={2}
          sm={4}
          xs={12}
        >
          <ForwardRef(Typography)
            className="makeStyles-contact-4"
          >
            @Place
          </ForwardRef(Typography)>
        </ForwardRef(Grid)>
      </ForwardRef(Grid)>
    </ForwardRef(Grid)>
  </div>
</div>
`;
